easybuild/easyblocks/c/cutest.py

```diff
@@ -0,0 +1,227 @@
+##
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing CUTEst, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+from distutils.version import LooseVersion
+import glob
+import os
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import apply_regex_substitutions, change_dir, remove_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd_qa
+
+
+COMPONENTS = ['ARCHDefs', 'CUTEst', 'SIFDecode']
+
+
+class EB_CUTEst(EasyBlock):
+    """Support for building/installing CUTEst."""
+
+    def __init__(self, *args, **kwargs):
+        """Constructor for CUTEst easyblock."""
+        super(EB_CUTEst, self).__init__(*args, **kwargs)
+
+        self.build_in_installdir = True
+
+        self.subdirs = {}
+
+        self.matlab_root = None
+
+    def prepare_step(self):
+        """Custom prepare step for CUTEst."""
+        super(EB_CUTEst, self).prepare_step()
+
+        # determine subdirectories for each component
+        for comp in COMPONENTS:
+            paths = glob.glob(os.path.join(self.installdir, comp + '-*'))
+            if len(paths) == 1:
+                self.subdirs[comp] = os.path.basename(paths[0])
+            else:
+                raise EasyBuildError("Failed to isolate %s subdirectory: %s", comp, paths)
```

Comment:
you still need to add the case for dry run here

Reply:
fixed in 849f9d9